kevm

```diff
@@ -103,21 +103,32 @@ run_kast() {
 }
 
 run_prove() {
-    local def_module run_dir proof_args bug_report_name
+    local def_module run_dir proof_args haskell_backend_command bug_report_name eventlog_name
 
     check_k_install
 
     def_module="$1" ; shift
     bug_report_name="kevm-bug-$(basename "${run_file%-spec.k}")"
+    eventlog_name="$run_file.eventlog"
+
+    haskell_backend_command=(kore-exec)
+    ! $bug_report || haskell_backend_command+=(--bug-report "$bug_report_name")
+    ! $profile    || haskell_backend_command+=(+RTS -l -ol$eventlog_name -RTS)
 
     proof_args=(--def-module "$def_module")
-    ! $debug                          || proof_args+=(--debug)
-    ! $debugger                       || proof_args+=(--debugger)
-    ! $bug_report                     || proof_args+=(--haskell-backend-command "kore-exec --bug-report $bug_report_name")
-    [[ ! -f ${concrete_rules_file} ]] || proof_args+=(--concrete-rules "$(cat ${concrete_rules_file} | tr '\n' ',')")
+    ! $debug                                   || proof_args+=(--debug)
+    ! $debugger                                || proof_args+=(--debugger)
+    [[ ${#haskell_backend_command[@]} -le 1 ]] || proof_args+=(--haskell-backend-command "${haskell_backend_command[*]}")
+    [[ ! -f ${concrete_rules_file} ]]          || proof_args+=(--concrete-rules "$(cat ${concrete_rules_file} | tr '\n' ',')")
 
     ! $debug || set -x
-    kprove --directory "$backend_dir" "$run_file" "${proof_args[@]}" "$@"
+    if [[ "$profile_timeout" != "0" ]]
+    then
+        timeout -s INT "$profile_timeout" kprove --directory "$backend_dir" "$run_file" "${proof_args[@]}" "$@" || true
+    else
+        kprove --directory "$backend_dir" "$run_file" "${proof_args[@]}" "$@" || true
+    fi
+    ! $profile || kore-prof "$eventlog_name" $kore_prof_args > "$eventlog_name.json"
```

Comment:
This should likely read (formatting and logic changes):

```suggestion
    if $profile && [[ "$profile_timeout" != "0" ]]; then
        timeout -s INT "$profile_timeout" kprove --directory "$backend_dir" "$run_file" "${proof_args[@]}" "$@" || true
        kore-prof "$eventlog_name" $kore_prof_args > "$eventlog_name.json"
    else
        kprove --directory "$backend_dir" "$run_file" "${proof_args[@]}" "$@"
    fi
```

Reply:
Initially I didn't have `|| true` on either branch and couldn't figure out why the timeout was killing the whole script (I'm clearly not used to bash 😅), then `$profile` in the condition didn't do anything. However this is both clearer and preserves the status code of `kprove` which is quite important.

However `kore-prof` needs to stay outside of the `if`, because it has to happen even if there is no timeout.

Reply:
I noticed that sometimes, if a `.json` file already existed at the expected place that weird things happened e.g. the new log got appended to the old `.json` content. This may be a user error, but to be safe, I suggest removing any old `.json` file that exists there before writing the one.

Reply:
That sounds like user error because the file is overwritten with the `>` redirect. Appending would happen if `>>` were used.

Are you sure you weren't putting log entries in a `.json` file (e.g with `--log [file] --log-entries [entries]`)? `kore-exec` appends those to the file, but that's not related to this PR.

Reply:
Ah, I was using an old version of your script where it wrote the log to the same file as ended up being the `.json` e.g.
```
! $profile || { kore-prof "$eventlog_name" > "$eventlog_name.tmp" \
               &&   mv "$eventlog_name.tmp" "$eventlog_name"; }
```
it looks like the latest version doesn't do that, so probably not an issue anymore